thrift/header_protocol.go

```diff
@@ -0,0 +1,305 @@
+/*
```

Comment:
is this file needed? I think the original copy of Thrift code was done with purging some of the unused parts

Reply:
Yeah, it's needed.

Reply:
I did the same as you did last time, copied the new thrift code from `apache/thrift` to `thrift/`, then add the missing files with needed interfaces and structs.